src/Block.php

```diff
@@ -91,21 +154,26 @@ public function render() {
       '#links' => $links,
       '#attributes' => array('class' => array('share-light')),
     );
+
     return $block;
   }
 
-  public static function from_current_path($options = array()) {
+  /**
+   * Populate the `$options` array with the current node's data.
+   */
+  public static function fromCurrentPath($options = array()) {
```

Comment:
Actually this creates a new `Block` instance. The `$options` parameter allows to pass additional settings, but `$options['node']` is overriden based on the current path. Also it may return `NULL` if the `toggle` is switched off in the node’s share_light configuration (the configuration is stored as a field item).